File: LeetCode/Tag/Binary-Search/cpp/704.binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int lo = 0, hi = nums.size()-1, mid;
        while (lo <= hi) {
            mid = lo + (hi-lo)/2;
            if (nums[mid] == target) return mid;
            if (nums[mid] < target)
                lo = mid+1;
            else
                hi = mid-1;
        }
        return -1;
    }
};

File: LeetCode/Tag/Binary-Search/js/704.binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
  let beg = 0, end = nums.length-1;
  let mid;
  while(beg <= end){
    mid = Math.floor((end+beg)/2);
    if(nums[mid] === target)
      return mid;
    else if(nums[mid] > target){
      end = mid-1;
    }
    else{
      beg = mid+1;
    }
  }
  return -1;
};
